artifact-version-diff/src/main/java/io/quarkus/qe/GeneratePomComparison.java

```diff
@@ -0,0 +1,142 @@
+import java.util.logging.Logger;
+
+public class GeneratePomComparison {
+
+    private final PomComparator pomComparator;
+    private final AllowedArtifacts allowedArtifactsFile;
+    private boolean allArtifactAreAllowed = true;
+    private boolean allArtifactVersionsAreAllowed = true;
+
+    public static final String HTML_BASE_START = """
+            <!DOCTYPE html>
+            <html>
+            <style>
+                table, td, th {
+                    border: 1px solid;
+                    vertical-align: top;
+                    text-align: left;
+                    padding: 5px;
+                }
+
+                table {
+                    width: 100%;
+                    border-collapse: collapse;
+                }
+                .not-allowed {
+                    background-color: #EE9999;
+                }
+            </style>
+            <body>
+            <div>""";
+
+    public static final String HTML_BASE_END = """
+            </div>
+            </body>
+            </html>""";
+    public String template = """
+                <tr>
+                    <td class="not-allowed">substitute-artifact</td>
+                </tr>
+                """;
+    private static final Logger LOG = Logger.getLogger(GeneratePomComparison.class.getName());
+
+    public GeneratePomComparison(AllowedArtifacts allowedArtifactsFile) throws IOException, XmlPullParserException {
+        pomComparator = new PomComparator();
+        pomComparator.comparePoms();
+        this.allowedArtifactsFile = allowedArtifactsFile;
+    }
+
+
+    /**
+     * Generate report of platform bom comparison of artifacts
+     * The output is saved to platformBomComparison.html
+     * There is need to check if `allowedArtifactsFile` is not null otherwise the exception is thrown.
+     */
+    public void generatePomComparisonReport() {
+        LOG.info("Generating platform bom comparison report");
+        try(FileWriter fw = new FileWriter("platformBomComparison.html")) {
+            fw.write(HTML_BASE_START);
+
+            fw.write("<h1>RHBQ missing artifacts</h1>\n<table>");
+            fw.write(generateDifferArtifacts(pomComparator.getMissingDependencies(),
+                    allowedArtifactsFile != null ? allowedArtifactsFile.getBomComparisonsMissingArtifacts() : null));
+            fw.write("</table>");
+
+            fw.write("<h1>RHBQ extra artifacts</h1>\n<table>");
+            fw.write(generateDifferArtifacts(pomComparator.getExtraDependencies(),
+                    allowedArtifactsFile != null ? allowedArtifactsFile.getBomComparisonsExtraArtifacts() : null));
+            fw.write("</table>");
+
+            fw.write("<h1>RHBQ artifacts with different version</h1>\n<table>");
+            fw.write(generateDifferentVersions(pomComparator.getMismatchedVersion(),
+                    allowedArtifactsFile != null ? allowedArtifactsFile.getVersionComparisonsArtifacts() : null));
+            fw.write("</table>");
```

Comment:
This should be dropped as it's duplicate for report we have in `outputDiff.html`